src/lib_json/json_tool.h

```diff
@@ -86,30 +88,25 @@ static inline void uintToString(LargestUInt value, char*& current) {
   } while (value != 0);
 }
 
-/** Change ',' to '.' everywhere in buffer.
+/** Change decimal point from current locale (i.e. ',') to '.' everywhere in buffer.
  *
  * We had a sophisticated way, but it did not work in WinCE.
  * @see https://github.com/open-source-parsers/jsoncpp/pull/9
  */
 template <typename Iter> Iter fixNumericLocale(Iter begin, Iter end) {
-  for (; begin != end; ++begin) {
-    if (*begin == ',') {
-      *begin = '.';
-    }
+  const char decimalPoint = getDecimalPoint();
+  if (decimalPoint != '\0' && decimalPoint != '.') {
+    std::replace(begin, end, decimalPoint, '.');
   }
-  return begin;
+  return end;
```

Comment:
compared to a previous implementation it has an undesired side-effect:
if used locale-independent version of jsoncpp, the former version would replace ',' with '.' if jsons happen to have ',' as a separator
with the change - it will not happen if current locale happened to be `C` or something
could be non-backward-compatible side-effect

Reply:
This private helper function is only ever used to correct the small buffers produced by `snprintf` of individual numeric values with either `%.*g` or `%.*f` format specifiers.

The idea was that `snprintf` is locale-dependent and JSON isn't, so we're trying to undo the possibly inappropriate decimal point. We don't have to worry about thousands separators, in particular, because we controlled the format string and didn't use a `'` modifier.

To really fix this problem we would have to synchronize the `getDecimalPoint()` call.
We should also make sure that it doesn't change between the time that we call snprintf and the time that we try to fix up the decimal points with this function. This is all kind of unappetizing work, but this PR isn't a valid shortcut IMO. We should instead change the snprintf calls to something that doesn't depend on locale in the first place.

Because we're a library we can't control the application locale. We really need to just use a different formatter.
I think using a `std::ostrstream` with `imbue` is probably the most portable way to go.

There's also `std::to_chars` when the platform is C++17 or beyond.

Reply:
> We should instead change the snprintf calls to something that doesn't depend on locale in the first place.

Sounds briliant.

> There's also `std::to_chars` when the platform is C++17 or beyond.

Tried that not so long ago: some compilers that mostly support C++17 might not have it, e.g. [libc++](https://libcxx.llvm.org/Status/Cxx17.html#id2) still has a partial support of it.